COMETwebapp/ViewModels/Pages/UserManagement/UserManagementPageViewModel.cs

```diff
@@ -0,0 +1,165 @@
+        /// </summary>
+        private readonly ISessionAnchor SessionAnchor;
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="UserManagementPageViewModel" /> class.
+        /// </summary>
+        /// <param name="sessionAnchor">The <see cref="ISessionAnchor" /></param>
+        public UserManagementPageViewModel(ISessionAnchor sessionAnchor)
+        {
+            this.SessionAnchor = sessionAnchor;
+        }
+
+        /// <summary>
+        ///     Tries to create a new <see cref="Person" />
+        /// </summary>
+        /// <returns>A <see cref="Task" /></returns>
+        public async Task Grid_ModelSaving()
+        {
+            var thingsToCreate = new List<Person>();
+            this.Person.EmailAddress.Add(this.EmailAddress);
+            this.Person.TelephoneNumber.Add(this.TelephoneNumber);
+            thingsToCreate.Add(this.Person);
+            
+            try
+            {
+                await this.SessionAnchor.CreateThingsSiteDirectory(thingsToCreate);
+            }
+            catch (Exception exception)
+            {
+                throw;
+            }        
+  
+            this.Person = new Person();
+            this.DataSource.Clear();
+            this.DataSource.AddRange(this.SessionAnchor.GetPersons());
+        }
+
+        /// <summary>
+        ///     Tries to deprecate a <see cref="Person" />
+        /// </summary>
+        /// <returns>A <see cref="Task" /></returns>
+        public async Task Grid_DataItemDeprecating(GridDataItemDeletingEventArgs e)
+        {
+            var personToDeprecate = new List<Person>();
+            var deprecatedPerson = (Person)e.DataItem;
+            var clonedPerson = deprecatedPerson.Clone(false);
+            clonedPerson.IsDeprecated = true;
+            personToDeprecate.Add(clonedPerson);
+            try
+            {
+                await this.SessionAnchor.UpdateThingsSiteDirectory(personToDeprecate);
+            }
+            catch (Exception exception)
+            {
+                throw;
+            }
+            this.DataSource.Clear();
+            this.DataSource.AddRange(this.SessionAnchor.GetPersons());
+        }
+
+        /// <summary>
+        ///     Method invoked when the component is ready to start, having received its
+        ///     initial parameters from its parent in the render tree.
+        ///     Override this method if you will perform an asynchronous operation and
+        ///     want the component to refresh when that operation is completed.
+        /// </summary>
+        /// <returns>A <see cref="Task" /> representing any asynchronous operation.</returns>
+        public void OnInitializedAsync()
+        {
+            this.DataSource.Clear();
+            this.DataSource.AddRange(this.SessionAnchor.GetPersons());
+            this.AvailableOrganizations = this.SessionAnchor.GetAvailableOrganizations();
```

Comment:
please do not use the intermediate ISEssionAnchor interface, just use the ISession directly